pkg/cfg/cfgmgr.go

```diff
@@ -163,71 +165,105 @@ func (c * CfgMgr) getRandomPassword() string {
     return string(out)
 }
 
-func (c *CfgMgr) CreateDB(serviceName, userName string) error {
+func (c *CfgMgr) CreateDB(serviceName, userName string) (bool, error) {
+    dbObject, err := c.getDbObject()
+    if err != nil {
+        return false, err
+    }
+    _, err = dbObject.Exec(fmt.Sprintf("SHOW DATABASE LIKE %s", serviceName))
+    if err == nil {
+        zap.L().Info(fmt.Sprintf("Database %s already exists", serviceName))
+        return false, nil
+    }
     zap.L().Debug("Creating DB for serviceName ", zap.String("serviceName", serviceName))
+    _, err = dbObject.Exec(fmt.Sprintf("CREATE DATABASE %s", serviceName))
+    if err != nil {
+        zap.L().Error("Error while creating database", zap.Error(err))
+        return false, err
+    }
+    zap.L().Info(fmt.Sprintf("Created DB '%s' successfully", serviceName))
+    return true, nil
+}
+
+func (c *CfgMgr) CreateGrants(serviceName, userName, dbPassword string) (bool, error) {
+    dbObject, err := c.getDbObject()
+    if err != nil {
+        return false, err
+    }
+    rows, err := dbObject.Query("SELECT @@hostname")
+    if err != nil {
+        zap.L().Error("Error while getting hostname", zap.Error(err))
+        return false, err
+    }
+    defer rows.Close()
+    var hostname string
+    updated := false
+    for rows.Next() {
+        _ = rows.Scan(&hostname)
+    }
+    hosts := []string{"localhost", "%", hostname}
+    for _, hostName := range hosts {
+        before_grants := c.getGrants(userName, hostName, dbObject)
+        query := fmt.Sprintf("GRANT ALL PRIVILEGES ON %s.* TO '%s'@'%s' IDENTIFIED BY '%s'",
+                             serviceName, userName, hostName, dbPassword)
+        _, _ = dbObject.Exec(query)
+        after_grants := c.getGrants(userName, hostName, dbObject)
+        if (reflect.DeepEqual(before_grants, after_grants)) {
+            updated = false
+        } else {
+            updated = true
+        }
+    }
+    return updated, nil
```

Comment:
If the last grant fails and all others are successful then this function will return `false`. Similarly if the grant fails for the first and second but succeeds for the last one, this function will return `true` incorrectly.